src/api/routes/render.ts

```diff
@@ -169,3 +188,66 @@ export async function renderJSON(
     res.status(500).json({ error: err.message });
   }
 }
+
+export async function processLogin(
+  req: express.Request,
+  res: express.Response
+): Promise<void> {
+  const { url, ua, username, password, renderHTML } = res.locals;
+  const headersToForward = getForwardedHeadersFromRequest(req);
+  try {
+    const {
+      error,
+      statusCode,
+      headers,
+      body,
+      cookies,
+      timeout,
+    } = await renderer.task({
+      type: 'login',
+      url,
+      headersToForward,
+      userAgent: ua,
+      login: {
+        username,
+        password,
+      },
+    });
+
+    if (error) {
+      if (renderHTML) {
```

Comment:
on the other `render()`, in case of error we don't send a 200 are you sure about this?

Reply:
By default we send a 400 here too.
It's only if asked explicitly (`renderHTML` in the request), i.e. for the test-website to render the page where we got the error (as there can be multiple steps, it can help to see where we are blocked at)